test/setup

```diff
@@ -0,0 +1,96 @@
+set -e
+
+
+exec {BASH_XTRACEFD}>&1
+echo "$0" "$@"
+set -x
+
+
+test -n "$MPV_MPRIS_TEST_NAME" ||
+exec "./env" "$0" "$@"
+
+
+test="$(basename "$0")"
+
+
+if [ "$test" = setup ] ; then
+	echo "This setup script should be sourced, not executed" >&2
+	exit 1
+fi
+
+
+input_json="$MPV_MPRIS_TEST_MPV_IPC/$test.mpv.ipc.input.json"
+output_json="$MPV_MPRIS_TEST_MPV_IPC/$test.mpv.ipc.output.json"
+ipc="$MPV_MPRIS_TEST_MPV_IPC/$test.mpv.ipc"
+log_prefix="$MPV_MPRIS_TEST_LOG/$test"
+mpv_log="$log_prefix.mpv.log"
+socat_log="$log_prefix.socat.log"
+file="$MPV_MPRIS_TEST_PLAY"
+
+
+prop () {
+	prop="$1"
+	shift
+	jo command[]=get_property command[]="$prop" "$@"
+}
+
+val () {
+	jq --exit-status "[.data==$1, .error==\"success\"] | all"
+}
+
+check () {
+	prop "$1" > "$input_json"
+	cat "$input_json"
+	rm -f "$socat_log"
+	# Pass the input JSON to socat and save the output JSON, also redirect
+	# socat errors to a file so we can check afterward if there are errors.
+	socat -lf"$socat_log" "$input_json" "$ipc" > "$output_json"
```

Comment:
This seems to be `socat` version dependent: Newer versions seem to assume bidirectional communication, even in case it is not useful (as here, when an input file is used). 
For me, using:
```
cat "$input_json" | socat -lf"$socat_log" - "$ipc" > "$output_json"
```
instead works fine.

Reply:
Actually my socat version also does what yours does, but the tests still passed because my jq expression was fine with an empty output JSON file. To get socat to do all the work itself, you need to use `OPEN:input` and `CREATE:output`, but those are more verbose than - which is a shortcut for STDIO, which is a shortcut for STDIN, STDOUT, which are shortcuts for `FD:0` and `FD:1`, or something, I can't get `OPEN`+`CREATE` to work. I'm going back to what you suggested, adding size checks for the output JSON and will try to figure out how to fix the jq expression.

Reply:
Found a workaround for the jq issue here: https://github.com/stedolan/jq/issues/1142